include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -156,14 +144,40 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 			const auto arrayLayers = state->inImage->getCreationParameters().arrayLayers;
 			static constexpr auto maxChannels = 4u;
 
-			uint8_t* const totalImageCacheScratch = state->imageTotalCacheScratch = state->scratchMemory + state->cachesOffset;
-			uint8_t* const totalImageCacheOutput = state->imageTotalCacheOutput = state->imageTotalCacheScratch + state->imageTotalScratchCacheByteSize;
-
 			#ifdef _IRR_DEBUG
 			memset(scratchMemory, 0, state->scratchMemoryByteSize);
 			#endif // _IRR_DEBUG
 
+			const core::vector3du32_SIMD scratchByteStrides = [&]()
+			{
+				const auto trueExtent = vector3du32_SIMD(state->extent.width, state->extent.height, state->extent.depth);
```

Comment:
common state (from which you inherit) has a union between `{extent,layerCount}` and a `vectorSIMDu32`

Reply:
corrected